server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -879,21 +888,27 @@ public void cleanup(
     private void createSnapshotsDeletion(
         Collection<SnapshotId> snapshotIds,
         long repositoryDataGeneration,
-        IndexVersion repositoryFormatIndexVersion,
+        IndexVersion minimumNodeVersion,
         ActionListener<SnapshotsDeletion> listener
     ) {
         if (isReadOnly()) {
             listener.onFailure(new RepositoryException(metadata.name(), "repository is readonly"));
         } else {
             threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(ActionRunnable.supply(listener, () -> {
                 final var originalRootBlobs = blobContainer().listBlobs(OperationPurpose.SNAPSHOT);
+                final var originalRepositoryData = safeRepositoryData(repositoryDataGeneration, originalRootBlobs);
```

Comment:
Currently, `repositoryFormatIndexVersion` is computed with `repositoryData` available in the caller. With this change, we call `safeRepositoryData` to get ahold of the `repositoryData`. I think this should be fine since the two `repositoryData` instances should have the exactly same `repositoryDataGeneration` and hence exactly the same content? But in that case, I am not quite sure why we need to load `repositoryData` here again though.

Reply:
We almost certainly don't re-load the `repositoryData` here, we'll just be using the cached version. The reason for calling `safeRepositoryData` here is really to do one final (best-effort) check for other clusters concurrently writing to the repository before starting the delete, because what we're about to do will corrupt their writes (in the old repo format at least).

Reply:
I added a comment about this in 6022172fc3a61664e79578a6bd0b5efa64e2c27f.